### PR TITLE
add WardOverview backend connection

### LINKED ISSUE #472: Unassign patient button on patient details view
The orange "Unassign"-Button should remove the assignment between a patient and the bed.

250274074-5c3b7f9c-095b-473e-8433-777bb00b971e

### LINKED ISSUE #474: Update closing behavior of the patient details view
> Clicking outside of any bed tiles in the left section closes the patient detail section and shows the patient list instead
Adding a close button to the top of the patient detail section

250274074-5c3b7f9c-095b-473e-8433-777bb00b971e

https://github.com/helpwave/design/issues/2

### LINKED ISSUE #477: Create and update room for ward via mutations

### LINKED ISSUE #478: Create patient on room overview

### LINKED ISSUE #482: Load rooms with corrosponding patient-bed assignment on room overview

### LINKED ISSUE #495: Rooms table on ward details view
### Describe the feature

Rooms table on ward details view

### Why is this feature needed?

we have a new table that should replace the tanstack package

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #477
closes #478
closes #482
closes #495
closes #472
closes #474

### CODE REVIEW

**Review by use-to — CHANGES_REQUESTED** [review]
Are these version pins related to #485 
As these changes aren't related to `tasks` we should think about moving that to #485 .

**Review by use-to — DISMISSED** [review]
@DasProffi could you check this pr again? I feel like there is a version mismatch with our `helpwave/ts-proto` resulting in loading issues.

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
- Room creations, updates and removes are not updating the UI on the ward form
- The patient details form does not switch into the creation mode and therefore a creation of a patient is not possible
- (The patient details of an already created patient does not display. I guess this is due to the new patient list. We should track this in a new issue)

**Comment by MaxSchaefer** [comment]
> @DasProffi could you check this pr again? I feel like there is a version mismatch with our `helpwave/ts-proto` resulting in loading issues.

@use-to the version is correct. The loading issue was due to https://github.com/helpwave/services/pull/253

**Comment by DasProffi** [comment]
@use-to Why did you mark it as draft when I request you rereview?

**Comment by DasProffi** [comment]
The rooms are updating but the current table doesn't allow for showing changes, this will be done in issue #495

**Comment by use-to** [comment]
> @use-to Why did you mark it as draft when I request you rereview?

It seems to me that editing the rooms does not work. So the connection in the WardOverview is not ready yet, right? If the PR is ready-for-review, I think that everything is ready and there is no more work on it, except the request changes.

**Comment by use-to** [comment]
> The rooms are updating but the current table doesn't allow for showing changes, this will be done in issue #495

Thats maybe the issue. We should fix that issue #495 first, then merge it into this branch and then set this PR to ready.

**Comment by MaxSchaefer** [comment]
@DasProffi @use-to this pr is getting too large and should be merged by the end of the day. I've updated #477, the logic for room creation and updates is working fine, just state is not propagated to the UI. This should be fixed by #495 and #496 directly after this pr.

Lets review and merge this.

**Comment by use-to** [comment]
> @DasProffi @use-to this pr is getting too large and should be merged by the end of the day. I've updated #477, the logic for room creation and updates is working fine, just state is not propagated to the UI. This should be fixed by #495 and #496 directly after this pr.
> 
> Let's review and merge this.

I disagree, but as we need progress - Go ahead.

**Comment by MaxSchaefer** [comment]
To test the underlying logic with an implemented UI, #495 and #496 will be added to this pull request. Because of that, I will move the pull request to draft.

**Comment by DasProffi** [comment]
@MaxSchaefer am I missing something that should be added with this PR or can I reopen it for review?

**Comment by MaxSchaefer** [comment]
@DasProffi looks good. You can reopen this.

File: tasks/components/layout/PatientDetails.tsx [hunk 1]
```diff
@@ -65,27 +69,51 @@ export const PatientDetail = ({
...
+  const [newPatient, setNewPatient] = useState<PatientDetailsDTO>(patient)
   const [newTask, setNewTask] = useState<TaskDTO | undefined>(undefined)
   const [boardObject, setBoardObject] = useState<KanbanBoardObject>({ searchValue: '' })
   const [isShowingSavedNotification, setIsShowingSavedNotification] = useState(false)
-  const [sortedTasks, setSortedTasks] = useState<SortedTasks>({
-    unscheduled: newPatient.tasks.filter(value => value.status === 'unscheduled'),
-    inProgress: newPatient.tasks.filter(value => value.status === 'inProgress'),
-    done: newPatient.tasks.filter(value => value.status === 'done'),
+
+  const sortedTasksByPatient = (patient: PatientDetailsDTO) => ({
```
Comment (MaxSchaefer) [inline tasks/components/layout/PatientDetails.tsx:90]:
```suggestion
  const sortedTasksByPatient = (patient: PatientDetailsDTO) => ({
```
Pass TaskDTO directly

File: tasks/components/layout/WardDetails.tsx [hunk 2]
```diff
@@ -37,7 +45,8 @@ const defaultWardDetailTranslations: Record<Languages, WardDetailTranslation> =
     deleteConfirmText: 'Do you really want to delete this ward?',
     deleteWard: 'Delete ward',
     create: 'Create',
-    update: 'Update'
+    update: 'Update',
+    roomsNotOnCreate: 'Rooms can only be added once the Ward is Created'
```
Comment (MaxSchaefer) [inline tasks/components/layout/WardDetails.tsx:49]:
```suggestion
    roomsNotOnCreate: 'Rooms can only be added once the ward is created'
```

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: tasks/mutations/ward_mutations.ts [hunk 3]
```diff
@@ -114,7 +133,7 @@ export const useWardQuery = (id: string) => {
   })
 }
 
-export const useUpdateMutation = (setSelectedWard: (ward:(WardDTO | undefined)) => void) => {
+export const useWardUpdateMutation = (callback: (ward:WardDTO) => void) => {
```
[the remark this answers is not in the document]
Comment (DasProffi) [reply tasks/mutations/ward_mutations.ts:137]: no, as these functions are only internal for the react query and provide no external access to the results, this a callback is required

**Comment by MaxSchaefer** [comment]
Looks good! @use-to did you plan to review as well?

**Review by MaxSchaefer — APPROVED** [review]

**Review by use-to — CHANGES_REQUESTED** [review]
[Screencast from 2023-07-10 22-52-30.webm](https://github.com/helpwave/web/assets/13922835/7c064bbb-0608-4b55-8b71-0faec20de870)

**Comment by use-to** [comment]
> Looks good! @use-to did you plan to review as well?

I did now, just dismiss my review if the requested change is not part of this pr.

**Comment by DasProffi** [comment]
> > Looks good! @use-to did you plan to review as well?
> 
> I did now, just dismiss my review if the requested change is not part of this pr.

yeah, this is not part of the pr for now, because the endpoint doesn't exist yet

**Pull request merged**